medium-tests/src/test/java/mediumtest/synchronization/PluginSynchronizationMediumTests.java

```diff
@@ -154,7 +154,7 @@ void it_should_not_pull_plugins_that_do_not_support_sonarlint() {
       .build(client);
 
     waitAtMost(3, SECONDS).untilAsserted(() -> {
-      assertThat(getPluginsStorageFolder()).doesNotExist();
+      assertThat(getPluginsStorageFolder().toFile().listFiles()).hasSize(1);
```

Comment:
I suggest checking a little bit further to verify whether the file's content is what we expect.

Reply:
Before this change, if no plugins were downloaded, there would be no file under the `/plugins` dir. Now, we create the `pb` file in any case, but there will be no plugin jar files. That's why I checked the file count.

I can also check the file name `plugin_references.pb`. I used to add this check first, but it looked cumbersome, so I removed it. If you think it is enough, I can add it.